greenplumpython/table.py

```diff
@@ -249,34 +246,17 @@ def assign(self, **new_columns: Dict[str, Callable[["Table"], Any]]) -> "Table":
 
         Returns:
             Table: New table including the new assigned columns
-
-        Warning:
-            Currently, value of type :class:`Expr` whose result has more than
-            one column is **not supported** even though no exception will
-            be thrown in that case. Please **don't rely on this behavior** as
-            this will be fixed soon.
-
-            Examples of this case include functions returning composite type
-            objects, etc.
-
-            This is because GreenplumPython have no knowledge on what columns
-            are in a :class:`Table` or the result of an :class:`Expr`. This
-            issue will be fixed as soon as we implement column inference for
-            GreenplumPython.
         """
         if len(new_columns) == 0:
             return self
-        new_columns_str = []
-        for k, f in new_columns.items():
-            v: Any = f(self)
-            if isinstance(v, Expr) and not (v.table is None or v.table == self):
-                raise Exception(
-                    "Current table and newly included expression must be based on the same table"
-                )
-            new_columns_str.append(
-                f"{v.serialize() if isinstance(v, Expr) else to_pg_const(v)} AS {k}"
-            )
-        return Table(f"SELECT *, {','.join(new_columns_str)} FROM {self.name}", parents=[self])
+        targets: List[str] = []
+        if len(new_columns):
+            for k, f in new_columns.items():
+                v: Any = f(self)
+                if isinstance(v, Expr) and not (v.table is None or v.table == self):
+                    raise Exception("Newly included columns must be based on the current table")
+                targets.append(f"{v.serialize() if isinstance(v, Expr) else to_pg_const(v)} AS {k}")
+            return Table(f"SELECT *, {','.join(targets)} FROM {self.name}", parents=[self])
```

Comment:
What if the user just want the result column of function and none of others, is this `*` necessary?